table/tables/cache.go

```diff
@@ -109,19 +106,21 @@ func (c *cachedTable) TryReadFromCache(ts uint64, leaseDuration time.Duration) k
 // newCachedTable creates a new CachedTable Instance
 func newCachedTable(tbl *TableCommon) (table.Table, error) {
 	ret := &cachedTable{
-		TableCommon: *tbl,
+		TableCommon:    *tbl,
+		lockingForRead: make(chan struct{}, 1),
+		renewReadLease: make(chan struct{}, 1),
 	}
 	return ret, nil
 }
 
 // Init is an extra operation for cachedTable after TableFromMeta,
 // Because cachedTable need some additional parameter that can't be passed in TableFromMeta.
-func (c *cachedTable) Init(renewCh chan func(), exec sqlexec.SQLExecutor) error {
-	c.renewCh = renewCh
+func (c *cachedTable) Init(exec sqlexec.SQLExecutor) error {
 	raw, ok := exec.(sqlExec)
 	if !ok {
 		return errors.New("Need sqlExec rather than sqlexec.SQLExecutor")
 	}
+	raw.ExecuteInternal(context.Background(), "set @@session.tidb_retry_limit = 0")
```

Comment:
We need to check the error to pass the linter.

Reply:
OK, I've removed this line...
Let it retry now, because without retry, there will be some unexpected [WARN] log...